doc/server/observability.md

```diff
@@ -0,0 +1,64 @@
+# Observability
+
+Metrics collection is possible by creating `Metric` instances and adding them to server options via `MetricsInterceptor`. 
+Certain endpoints can be ignored by adding their definitions to `ignoreEndpoints` list.
+`Metric` wraps an aggregation object (like counter or gauge), and requires a function
+returning `EndpointMetric` with metric collection logic. There are three callbacks in `EndpointMetric`:
+
+1. `onRequest` - called after successful request decoding
```

Comment:
would be good to describe the two-stage `onRequest` (with metric initialisation)